### PR TITLE
create a meta.yaml file to create a conda package for flytekit on condo-forge

### PR DESCRIPTION
**Pull request by Qiwen-Yu**
# TL;DR
_Please replace this text with a description of what this PR accomplishes._

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This is the meta.yaml file required to publish a conda package. Part of it was edited manually, including `noarch: python`, `python >=3.6` in host and run, and the `about` and `extra` section. This requires reviews before submit to https://github.com/conda-forge/staged-recipes as a new recipe.

Ref: https://conda-forge.org/#contribute

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1679

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by Qiwen-Yu** [comment]
In this error, https://github.com/flyteorg/flytekit/runs/4057226573?check_suite_focus=true, this code 
https://github.com/flyteorg/flytekit/runs/4057226573?check_suite_focus=true#step:6:332 was auto-generated and should not be changed, according to conda-forge

File: docs/source/design/authoring.rst [hunk 1]
```diff
@@ -114,15 +114,15 @@ Before going further, there is a special object that's worth mentioning, the :py
 Let's assume we have a workflow like ::
```
Comment (kumare3) [inline docs/source/design/authoring.rst:114]: wrong PR?

File: meta.yaml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    - flytekit.common
+    - flytekit.common.core
+    - flytekit.common.exceptions
+    - flytekit.common.mixins
+    - flytekit.common.tasks
+    - flytekit.common.tasks.sagemaker
+    - flytekit.common.types
+    - flytekit.common.types.impl
+    - flytekit.configuration
+    - flytekit.contrib
```
Comment (kumare3) [inline meta.yaml:79]: actually we can drop some of the packages. Refer to - https://docs.flyte.org/projects/flytekit/en/latest/contributing.html#quick-background

File: meta.yaml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    - pyflyte=flytekit.clis.sdk_in_container.pyflyte:main
+    - flyte-cli=flytekit.clis.flyte_cli.main:_flyte_cli
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - checksumdir >=1.2.0
```
Comment (kumare3) [inline meta.yaml:28]: is there no way to re-use the requirements? I prefer not to duplicate it

**Comment by kumare3** [comment]
@Qiwen-Yu thank you so much for opening the PR. Added a couple comments

File: meta.yaml (hunk 3 above)
Comment (Qiwen-Yu) [reply meta.yaml:28]: I am afraid that the requirements.txt can not be reused here. This `meta.yaml` is the only file needed when submitting to https://github.com/conda-forge/staged-recipes.
However, I am not sure whether it can be simplified or not. I am looking at [examples](https://github.com/conda-forge/staged-recipes/pulls?q=is%3Apr+is%3Aclosed) here, which are closed PRs at conda-forge.
For instance, this one https://github.com/conda-forge/staged-recipes/pull/16139/files, duplicate its own requirements.txt in https://github.com/coiled/dask-snowflake

File: meta.yaml (hunk 2 above)
Comment (Qiwen-Yu) [reply meta.yaml:79]: I will edit this section.

**Comment by cosmicBboy** [comment]
note: the `meta.yaml` file in this PR will not stay in the flytekit repo for long. Once this PR is merged, we need to follow steps in this guide: https://conda-forge.org/#add_recipe. Basically we need to copy our `meta.yaml` file over to the flyteorg fork of https://github.com/conda-forge/staged-recipes, which is where we submit this recipe into the `conda-forge` channel infrastructure. Once our PR is merged there, there's no need to maintain a `meta.yaml` file in the flytekit repo.

**Comment by wild-endeavor** [comment]
hey @Qiwen-Yu did you still want to work on this PR?  Or should we push this through?

Thanks!

**Comment by Qiwen-Yu** [comment]
@wild-endeavor @cosmicBboy  I deleted some packages in `test imports` section addressing the [comment](https://github.com/flyteorg/flytekit/pull/731#discussion_r739748620) from @kumare3. By the way, the Python version requirement in the meta.yaml is now set as `>= 3.6`, should I change it to ` >= 3.7`?

**Comment by eapolinario** [comment]
flytekit is a conda package [now](https://libraries.io/conda/flytekit).
